mdata/aggmetric.go

```diff
@@ -409,14 +410,23 @@ func (a *AggMetric) Add(ts uint32, val float64) {
 		a.add(ts, val)
 	} else {
 		// write through reorder buffer
-		res, accepted := a.rob.Add(ts, val)
-
-		if len(res) == 0 && accepted {
-			a.lastWrite = uint32(time.Now().Unix())
-		}
-
-		for _, p := range res {
-			a.add(p.Ts, p.Val)
+		res, err := a.rob.Add(ts, val)
+
+		if err == nil {
+			if len(res) == 0 {
+				a.lastWrite = uint32(time.Now().Unix())
+			} else {
+				for _, p := range res {
+					a.add(p.Ts, p.Val)
+				}
+			}
+		} else {
+			var reason string
```

Comment:
This needs to be initialised to something, eg "unknown".

Reply:
fixed